server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -354,20 +353,23 @@ func validateOutgoingOAuthConnectionCredentials(c *Context, w http.ResponseWrite
 	// Allow checking connections sent in the body or by id if coming from an already existing
 	// connection url.
 	var inputConnection *model.OutgoingOAuthConnection
-	if c.Params != nil && c.Params.OutgoingOAuthConnectionID != "" {
+	if err := json.NewDecoder(r.Body).Decode(&inputConnection); err != nil {
```

Comment:
We need to use LimiterReader here, otherwise a really large payload could be sent (e.g. in the name) and cause performance/availability issues.  We can use the *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes for the max size.

Please speak with @sbishel or @harshilsharma63 as they are currently working on something related and a unified approach is needed. Example: https://github.com/mattermost/mattermost/pull/25510/files

Reply:
Added `io.LimitedReader` to secure payload size here: https://github.com/mattermost/mattermost/pull/25507/commits/8ea552a82e99a14d40b8335c16e796706226686d

The provided solution was for more generic payloads, but the approach here is the same as they use in that PR.